ow_power_system/src/PowerSystemPack.cpp

```diff
@@ -167,22 +54,21 @@ void PowerSystemPack::InitAndRun()
   // Initialize EoD_events and previous_times.
   for (int i = 0; i < NUM_NODES; i++)
   {
-    previous_times[i] = 0;
-    for (int j = 0 ; j < 3; j++)
-    {
-      EoD_events[i][j] = -1;
-    }
+    m_nodes[i].previous_time = 0;
+    EoD_events[i].remaining_useful_life = -1;
+    EoD_events[i].state_of_charge = -1;
+    EoD_events[i].battery_temperature = -1;
   }
 
   // Construct the prediction handlers.
-  PredictionHandler handler_0(bus[0], node_names[0]);
-  PredictionHandler handler_1(bus[1], node_names[1]);
-  PredictionHandler handler_2(bus[2], node_names[2]);
-  PredictionHandler handler_3(bus[3], node_names[3]);
-  PredictionHandler handler_4(bus[4], node_names[4]);
-  PredictionHandler handler_5(bus[5], node_names[5]);
-  PredictionHandler handler_6(bus[6], node_names[6]);
-  PredictionHandler handler_7(bus[7], node_names[7]);
+  PredictionHandler *handlers[NUM_NODES];
+  for (int i = 0; i < NUM_NODES; i++)
+  {
+    handlers[i] = new PredictionHandler(EoD_events[i].remaining_useful_life,
+                                        EoD_events[i].state_of_charge,
+                                        EoD_events[i].battery_temperature,
+                                        m_nodes[i].bus, m_nodes[i].name, i);
+  }
```

Comment:
[required]
For each use of `new` you should plan to also call `delete` when the object is no longer needed, _i.e._ in the destructor. Otherwise you have a memory leak.

If the elements must be pointers, then I recommend using `std::unique_ptr`, which will delete the objects automatically when the pointer goes out of scope (no `delete` call needed). If the elements don't need to be pointers, then you could construct them in-place, although this would require a dynamically allocated container, such as `std::vector`.

There are different acceptable approaches here, but the **required** part of this comment is that we fix the memory leak. Here's just one option that both maintains the pointer-ness of the elements and the static-ness of their container. 
```suggestion
  auto handlers = std::array<std::unique_ptr<PredictionHandler>, NUM_NODES>();
  for (int i = 0; i < NUM_NODES; i++)
  {
    handlers[i] = std::make_unique<PredictionHandler>(
      EoD_events[i].remaining_useful_life,
      EoD_events[i].state_of_charge,
      EoD_events[i].battery_temperature,
      m_nodes[i].bus, m_nodes[i].name, i
    );
  }
```
`std::array` is just a modern C++ wrapper for old C++ arrays that enables the same methods other modern C++ containers have. Using either is fine here. They are essentially the same.